Modules/Core/Traits/HandlesDeletion.php

```
@@ -0,0 +1,115 @@
+        ?string $successMessage = null,
+        ?string $errorMessage = null
+    ): RedirectResponse {
+        try {
+            // Execute the delete operation
+            $result = $deleteCallback();
+
+            // Early return on failure
+            if ($result === false) {
+                return $this->redirectWithError(
+                    $redirectRoute,
+                    $errorMessage ?? TranslationHelper::trans('record_deletion_failed')
```

PHPMD (2.15.0)

Avoid using static access to class '\Modules\Core\Support\TranslationHelper' in method 'executeDelete'. (undefined)

(StaticAccess)